lang/java/avro/src/main/java/org/apache/avro/reflect/ReflectData.java

```diff
@@ -737,6 +737,9 @@ protected Schema createSchema(Type type, Map<String, Schema> names) {
 
               AvroName annotatedName = field.getAnnotation(AvroName.class); // Rename fields
               String fieldName = (annotatedName != null) ? annotatedName.value() : field.getName();
+              if ("this$0".equals(fieldName)) {
```

Comment:
Can you split the string out to a static private final variable?

Reply:
@Fokko any clear reason? It seems to me that there's no problem with performance: https://stackoverflow.com/questions/53835748/java-performance-private-static-final-string-vs-local-string-number-of-objec/53837197

Reply:
For readability. By splitting out this constant we can name the variable: https://en.wikipedia.org/wiki/Magic_number_(programming)

Reply:
@Fokko agree. Do you have any suggestion for the constant name? Is `STRING_OUTER_PARENT_REFERENCE` ok?

Reply:
Yes, that sounds good!

Reply:
@Fokko I've fixed. :)